source/function.cpp

```diff
@@ -40,40 +40,54 @@ using namespace fmt::literals;
 
 namespace binder {
 
-static std::map<string, string > const cpp_python_operator_map{
-	{"operator+", "__add__"}, //
-	{"operator-", "__sub__"}, //
-	{"operator*", "__mul__"}, //
-	{"operator/", "__truediv__"}, //
-	{"operator%", "__mod__"}, //
-	{"operator~", "__invert__"}, //
-	{"operator|", "__or__"}, //
-	{"operator&", "__and__"}, //
-	{"operator^", "__xor__"}, //
-	{"operator<<", "__lshift__"}, //
-	{"operator>>", "__rshift__"}, //
-
-	{"operator+=", "__iadd__"}, //
-	{"operator-=", "__isub__"}, //
-	{"operator*=", "__imul__"}, //
-	{"operator/=", "__itruediv__"}, //
-	{"operator%=", "__imod__"}, //
-	{"operator|=", "__ior__"}, //
-	{"operator&=", "__iand__"}, //
-	{"operator^=", "__ixor__"}, //
-	{"operator<<=", "__ilshift__"}, //
-	{"operator>>=", "__irshift__"}, //
-
-	{"operator()", "__call__"}, //
-	{"operator==", "__eq__"}, //
-	{"operator!=", "__ne__"}, //
-	{"operator[]", "__getitem__"}, //
-	{"operator=", "assign"}, //
-	{"operator++", "plus_plus"}, //
-	{"operator--", "minus_minus"}, //
-
-	{"operator->", "arrow"}, //
-};
+// Return the python operator that maps to the C++ operator; returns "" if no mapping exists
+string cpp_python_operator(const FunctionDecl & F) {
+	static std::map<string, vector<string>> const m {
+		{"operator+", {"__pos__", "__add__"}}, //
+		{"operator-", {"__neg__", "__sub__"}}, //
+		{"operator*", {"dereference", "__mul__"}}, //
+		{"operator/", {"__truediv__"}}, //
+		{"operator%", {"__mod__"}}, //
+		{"operator~", {"__invert__"}}, //
+		{"operator|", {"__or__"}}, //
+		{"operator&", {"__and__"}}, //
+		{"operator^", {"__xor__"}}, //
+		{"operator<<", {"__lshift__"}}, //
+		{"operator>>", {"__rshift__"}}, //
+
+		{"operator+=", {"__iadd__"}}, //
+		{"operator-=", {"__isub__"}}, //
+		{"operator*=", {"__imul__"}}, //
+		{"operator/=", {"__itruediv__"}}, //
+		{"operator%=", {"__imod__"}}, //
+		{"operator|=", {"__ior__"}}, //
+		{"operator&=", {"__iand__"}}, //
+		{"operator^=", {"__ixor__"}}, //
+		{"operator<<=", {"__ilshift__"}}, //
+		{"operator>>=", {"__irshift__"}}, //
+
+		{"operator()", {"__call__"}}, //
+		{"operator==", {"__eq__"}}, //
+		{"operator!=", {"__ne__"}}, //
+		{"operator[]", {"__getitem__"}}, //
+		{"operator=", {"assign"}}, //
+		{"operator++", {"plus_plus"}}, //
+		{"operator--", {"minus_minus"}}, //
+
+		{"operator->", {"arrow"}} //
+  };
+	const auto & found = m.find(F.getNameAsString());
+	if (found != m.end()) {
+		const auto & vec { found->second };
+		if (vec.size() == 1) { return vec[0]; }
+		const auto n = F.getNumParams();
+		if (vec.size() > n) {
+			return vec[n];
+		}
```

Comment:
That will work as well, before if you wanted a catch all you would just put it as the sole argument but I can update it to this. I don't think it would make a difference since I don't recall C++ allowing additional operator parameters besides the ones I coded, but I suppose this is safer in case I missed anything.